### PR TITLE
Fix for issue #759.

### PR DESCRIPTION
**Pull request by ILMTitan**
Fix for issue #759.
Add unit tests for ResourceManagerDataSource.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/DataSourceUnitTestsBase.cs [hunk 1]
```diff
@@ -81,6 +84,44 @@ public class DataSourceUnitTestsBase
...
+        /// </param>
+        /// <param name="requestMethod">
+        ///     The request expression on the resource (e.g. Topics => Topics.List(It.IsAny&lt;string&gt;()))
+        /// </param>
+        /// <param name="parameterValues">
+        ///     Parameter values to pretend the method was given.
+        /// </param>
+        /// <param name="responses">The list of reponses for the request to return.</param>
+        /// <returns>A mocked version of the service.</returns>
+        protected static S GetMockedService<S, P, RequestType, ResponseType>(
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/DataSourceUnitTestsBase.cs]: use name like:  TService, TResourceType, TRequestType, TResponseType

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
+﻿using Google.Apis.CloudResourceManager.v1;
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: Add the license text.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        private static readonly Project s_disabledProject = new Project
+        {
+            Name = DisabledProjectName,
+            ProjectId = DisabledProjectId,
+            LifecycleState = ResourceManagerDataSource.LifecycleState.DeleteRequested
+        };
+
+        [TestMethod]
+        [ExpectedException(typeof(DataSourceException))]
+        public async Task GetProjectsListAsync_Exception()
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: nit: Remove underscore in name.  It's not our naming convention.
Same as the following

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            var responses = new ListProjectsResponse[0];
+            CloudResourceManagerService service = GetMockedService(
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: How does this trigger GoogleApiException  ?   May you add comment?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                (CloudResourceManagerService s) => s.Projects, p => p.List(), new object[0], responses);
+            var dataSource = new ResourceManagerDataSource(null, init => service, null);
+
+            await dataSource.GetSortedActiveProjectsAsync();
+        }
+
+        [TestMethod]
+        public async Task GetSortedActiveProjectsAsync_Success()
+        {
+            var responses = new[]
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: nit: Create a static responses or helper method that it is reused in multiple tests.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/DataSourceUnitTestsBase.cs (hunk 1 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/DataSourceUnitTestsBase.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs (hunk 2 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs (hunk 3 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs (hunk 4 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs (hunk 5 above)
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources.UnitTests/ResourceManagerDataSourceUnitTests.cs]: I disagree. 

1. There is no reason these should all be the same
2. The methods are already very short and fit in a single screen.
3. Someone looking at the test needs to know the mock server responses to understand the test.

**Pull request merged**